projects/br_scs/src/Mirza/BusinessRegistry/API.hs

```diff
@@ -74,13 +74,10 @@ type PublicAPI =
 
 
 type PrivateAPI =
-       -- TODO: Note the following end point does't really make sense any more, it has only been left in place for now
-       --       because the plan is to basically refactor it into an associate user with a business end point.
-       "user"      :> "add"      :> ReqBody '[JSON] NewUser     :> Post '[JSON] UserId
-  :<|> "business"  :> "add"      :> ReqBody '[JSON] NewBusiness :> Post '[JSON] GS1CompanyPrefix
-  :<|> "key"       :> "add"      :> ReqBody '[JSON] JWK         :> QueryParam "expirationTime" ExpirationTime :> Post '[JSON] BRKeyId
-  :<|> "key"       :> "revoke"   :> Capture "keyId" BRKeyId     :> Post '[JSON] RevocationTime
-  :<|> "location"  :> "add"      :> ReqBody '[JSON] NewLocation :> Post '[JSON] LocationId
-  :<|> "company"   :> Get '[JSON] [BusinessResponse]
-
-
+       "user"          :> "add"      :> Put '[JSON] NoContent
+  :<|> "organisation"  :> "add"      :> ReqBody '[JSON] NewBusiness             :> Post '[JSON] GS1CompanyPrefix
+  :<|> "organisation"  :> Capture "gs1CompanyPrefix" GS1CompanyPrefix :> "user" :> Capture "userId" UserId                    :> Put '[JSON] NoContent
```

Comment:
maybe rename this to `/add/user/organisation`?
Just `/organisation` may not be very intuitive.

Reply:
So from Alex original design the plan was always noun / verb so user/add business/add, business/list. I'm happy with whatever you and @paulsamways think here as I haven't really had experience in what is nice and clean in terms of API and will build my experience and opinion over time.

Reply:
REST API's are don't typically have verbs, resource-orientated approach is to refer to either a collection (e.g. organisations), or a single entity (e.g. organisations/<id>).

Below was just an example I was discussing with Ashley in person, see https://restfulapi.net/resource-naming/ or look at Github API (https://developer.github.com/v3/users/) for more info.

-   /users/<id> - PUT, create -> {id, email}
/users/<id> - POST, update
/users/<id> - GET fetch

-   /users - POST { email }  -> { id, email, url: /users/<id> }
/users - GET fetch all

Reply:
So we have:
```
[PUBLIC]
GET /healthz
GET /version
GET /key/get/{keyId}
GET /key/getInfo/{keyId}
GET /organisation/search
GET /location/get/{GLN}
GET /location/search
GET /prototype/location/ux
GET /prototype/location/uxgln/{GLN}/{gs1companyprefix}

[PRIVATE]
PUT /user/add
GET /user/listorganisations
POST /organisation/add
PUT /organisation/{gs1CompanyPrefix}/user/{userId}
POST /key/add
POST /key/revoke/{keyId}
POST /location/add
```

Does this feel right?
```
[PUBLIC]
GET /healthz
GET /version
GET /key/{keyId}
GET /key/info/{keyId}
GET /organisation/search
GET /location/{GLN}
GET /location/search
[Ignoring the prototype functions for now]

[PRIVATE]
PUT /user/
GET /user/{id}/organisations/
PUT /organisation/{gs1CompanyPrefix}
PUT /organisation/{gs1CompanyPrefix}/member/{userId}
POST /key
POST /key/revoke/{keyId}
POST /location/add
```

Beyond `PUT /user/`, `GET /user/{id}/organisations/` and `PUT /organisation/{gs1CompanyPrefix}/member/{userId}` this is really out of scope for this PR and would suggest breaking it out into another issue / PR (since the change is more then a refactor of names and endpoints, because the types really need to have the id removed from them so that its not duplicated and error prone between for instance `/organisation/{gs1CompanyPrefix}` and 
```
{
  "newBusinessGS1CompanyPrefix": {
    "unGS1CompanyPrefix": "string"
  },
  "newBusinessName": "string",
  "newBusinessUrl": "string"
}
```
but good to have a master plan to work to now so that can make these endpoints line up with what the final design will be.

Reply:
```
[PUBLIC]
GET /healthz
GET /version
GET /keys/{keyId}         - returns full object (meta + jwk)
GET /keys/{keyId}/jwk   - returns just the jwk of the object
GET /organisations        - use query params for filters
GET /locations/{GLN}
GET /locations                - use query params for filters (no params means return all form collection)
[Ignoring the prototype functions for now]

[PRIVATE]
PUT /user                           - special singular case becaue `user` represents yourself (singular), implicit ID
GET /user/organisations/    
PUT /organisation/{gs1CompanyPrefix}
PUT /organisation/{gs1CompanyPrefix}/member/{userId}
POST /keys                           - adds a key
POST /keys/{keyId}/revoke   - revoke a key (breaks resource-orientated approach)
POST /locations                    - adds a location
```